book/src/commands.md

```diff
@@ -1,11 +1,89 @@
 # Commands
 
 - [Typable commands](#typable-commands)
+  - [Command mode syntax](#command-mode-syntax)
+    - [Quoting](#quoting)
+    - [Expansions](#expansions)
+    - [Flags](#flags)
+    - [Exceptions](#exceptions)
+  - [Built-ins](#built-ins)
 - [Static commands](#static-commands)
 
 ## Typable commands
 
-Typable commands are used from command mode and may take arguments. Command mode can be activated by pressing `:`. The built-in typable commands are:
+Typable commands are used from command mode and may take arguments. Command mode can be activated by pressing `:`.
+
+### Command mode syntax
+
+Command mode has rules for parsing the command line to evaluate quotes and expansions and split the line into positional arguments and flags. Most commands use these rules but some commands have custom parsing rules (see [Exceptions](#exceptions) below).
```

Comment:
I took a look at these docs since this PR is looking pretty close, and I want to suggest moving the details about parsing down so it's not the first thing someone sees on this page. I've been using this feature for a long time and it takes some effort for me to read through this and understand it. Maybe move the entire `Command mode syntax` section down below `Built-ins`? 

```
- [Built-ins](#built-ins)
- [Command mode syntax](#command-mode-syntax)
  - [Quoting](#quoting)
  - [Expansions](#expansions)
  - [Flags](#flags)
  - [Exceptions](#exceptions)
```

I would guess most people coming to this page just want to see the list of commands.

Reply:
Maybe it's better to move all of these docs out of `commands.md` since that doc is already really long (because of the generated part). All of this command mode usage stuff could be a new section under usage instead.